framework/CodeInterfaces/Saphire/SaphireInterface.py

```diff
@@ -0,0 +1,163 @@
+      for elem in elems:
+        ext = elem[0]
+        if ext not in usedExt:
+          usedExt.append(ext)
+        else:
+          raise IOError('Saphire cannot handle multiple input files with the same extension')
+        found=False
+        for inf in inputFiles:
+          if '.'+inf.getExt() == ext:
+            found=True
+            inFiles.remove(inf)
+            break
+        if not found:
+          raise IOError('input extension "'+ext+'" listed in input but not in inputFiles!')
+
+    #TODO if any remaining, check them against valid inputs
+
+    #PROBLEM this is limited, since we can't figure out which .xml goes to -i and which to -d, for example.
+    def getFileWithExtension(fileList,ext):
+      """
+      Just a script to get the file with extension ext from the fileList.
+      @ In, fileList, the string list of filenames to pick from.
+      @ Out, ext, the string extension that the desired filename ends with.
+      """
+      found = False
+      for index,inputFile in enumerate(fileList):
+        if inputFile.getExt() == ext:
+          found=True
+          break
+      if not found:
+        raise IOError('No InputFile with extension '+ext+' found!')
+      return index,inputFile
+
+    todo = ''
+    todo += executable
+    index=None
+    #inputs
+    for flag,elems in clargs['input'].items():
+      if flag == 'noarg': continue
+      todo += ' '+flag
+      for elem in elems:
+        ext, delimiter = elem[0], elem[1]
+        idx,fname = getFileWithExtension(inputFiles,ext.strip('.'))
+        todo += delimiter + fname.getAbsFile()
+        if index == None:
+          index = idx
+    #outputs
+    self.caseName = inputFiles[index].getBase()
+    outFile = 'out~'+self.caseName
+    if self.fixedOutFileName is not None:
+      outFile = self.fixedOutFileName
+    returnCommand = [('parallel',todo)],outFile
+    print('Execution Command: '+str(returnCommand[0]))
+    return returnCommand
+
+  def finalizeCodeOutput(self,command,output,workingDir):
+    """
+      This method is called by the RAVEN code at the end of each run. It will convert the SAPHIRE outputs into RAVEN
+      compatible csv file.
+      @ In, command, string, the command used to run the just ended job
+      @ In, output, string, the Output name root
+      @ In, workingDir, string, current working dir
+    """
+    filesIn = []
+    for outName, outType in self.codeOutputs.items():
+      filesIn.append((os.path.join(workingDir, self.outputDest, outName), outType))
+    outputParser = SaphireData(filesIn)
+    outputParser.writeCSV(os.path.join(workingDir, output))
+
+  def checkForOutputFailure(self,output,workingDir):
+    """
+      This method is called by RAVEN at the end of each run if the return code is == 0.
+      @ In, output, string, the Output name root
+      @ In, workingDir, string, current working dir
+      @ Out, failure, bool, True if the job is failed, False otherwise
+    """
+    failure = False
+    return failure
```

Comment:
mmm... No way to check for the failure?

Reply:
Currently, I do not know it. When there is a problem in the code runs, there will be some message in the SAPHIRE GUI, but the program will not exit. So basically, I do not know.